### PR TITLE
Feature/dab canisters info

### PR DESCRIPTION
**Pull request by maksym-boytsov**
## Changelog

- Bumped `plug-inpage-provider` and `plug-controller` versions
- Added functions for ICRocks-based URLs

### Type of changes included:

- [X] Components
- [X] Business Logic
- [X] Bug fixes
- [X] Styling
- [X] Code Refactor

### Clubhouse Tickets Related:

- https://app.shortcut.com/terminalsystems/story/24040/v4-get-canister-names-on-connect-modal
- https://app.shortcut.com/terminalsystems/story/24041/v4-get-canister-names-on-apps-tab

### Screenshots/Gifs:

Screenshot 2021-09-29 at 18 07 17
Screenshot 2021-09-29 at 18 08 54

### Tested on:

- [X] Chrome
- [X] Firefox
- [ ] Safari

### CODE REVIEW

File: source/Background/Controller.js [hunk 1]
```diff
@@ -120,6 +136,10 @@ backgroundController.exposeController(
 
     // if we receive a whitelist, we create agent
     if (isValidWhitelist) {
+      const canistersInfo = await fetchCanistersInfo(whitelist);
+
+      console.log('Canister info == ', canistersInfo);
```
Comment (maksym-boytsov) [inline source/Background/Controller.js:141]: For some reason it's always `undefined` 🤕 
In a process of figuring out what's going on

Comment (rocky-fleek) [reply source/Background/Controller.js:141]: @aleacevedo @Nima-Ra any idea?

File: source/Background/Controller.js [hunk 2]
```diff
@@ -29,6 +29,32 @@ const backgroundController = new BackgroundController({
...
+        };
+      }
+
+      return { id };
+    }));
+
+    const sortedCanistersInfo = canistersInfo.sort((a, b) => {
+      if (a.name && !b.name) return -1;
+      return 1;
+    });
```
Comment (rocky-fleek) [inline source/Background/Controller.js:50]: I thing you could do ```... => !a.name || b.name```

File: source/Background/Controller.js [hunk 3]
```diff
@@ -29,6 +29,32 @@ const backgroundController = new BackgroundController({
...
+
+      if (canisterInfo) {
+        return {
+          id,
+          ...canisterInfo,
+        };
+      }
+
+      return { id };
+    }));
```
Comment (rocky-fleek) [inline source/Background/Controller.js:45]: we should add a try catch in case dab fails

File: source/Background/index.js [hunk 4]
```diff
@@ -4,11 +4,11 @@ import { init } from './Controller';
...
   /* eslint-disable-next-line no-console */
-  console.log('controller started');
+  console.log('controller instantiated on install');
 });
 
 extension.runtime.onStartup.addListener(async () => {
   await init();
   /* eslint-disable-next-line no-console */
-  console.log('controller started');
+  console.log('controller instantiated on startup');
```
Comment (rocky-fleek) [inline source/Background/index.js:13]: nice ones

File: source/ui/CanisterInfo/components/Container/styles.js [hunk 5]
```diff
@@ -1,12 +1,13 @@
...
-  whitelistContainer: {
+  canisterInfoContainer: {
     display: 'flex',
     width: '100%',
     flexDirection: 'column',
-    marginBottom: 24,
-    maxHeight: 173,
-    overflow: 'auto',
+    overflowY: 'auto',
+    overflowX: 'hidden',
```
Comment (rocky-fleek) [inline source/ui/CanisterInfo/components/Container/styles.js:9]: why hide it?

File: source/ui/CanisterInfo/components/Container/index.jsx [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+
+  return <div ref={ref} className={clsx(classes.canisterInfoContainer, className)} {...props} />;
+});
+
+export default CanisterInfoContainer;
+
+CanisterInfoContainer.displayName = 'CanisterInfoContainer';
+
+CanisterInfoContainer.defaultProps = {
+  className: undefined,
```
Comment (rocky-fleek) [inline source/ui/CanisterInfo/components/Container/index.jsx:17]: this should be `''`

**Review by rocky-fleek — COMMENTED** [review]
Looks pretty good overall, just some minor details. I'd like to add the try catch if possible tho

File: source/ui/CanisterInfo/components/Container/styles.js (hunk 5 above)
Comment (maksym-boytsov) [reply source/ui/CanisterInfo/components/Container/styles.js:9]: expand icon button kind of extends the width over the 100%, and scrollbar appears

File: source/ui/CanisterInfo/components/Container/index.jsx (hunk 6 above)
Comment (maksym-boytsov) [reply source/ui/CanisterInfo/components/Container/index.jsx:17]: right, thanks

File: source/Background/Controller.js (hunk 3 above)
Comment (maksym-boytsov) [reply source/Background/Controller.js:45]: good catch, will do before release

File: source/Background/Controller.js [hunk 7]
```diff
@@ -29,6 +29,37 @@ const backgroundController = new BackgroundController({
...
+    try {
+      const canistersInfo = await Promise.all(whitelist.map(async (id) => {
+        const canisterInfo = await PlugController.getCanisterInfo(id);
+
+        if (canisterInfo) {
+          return {
+            id,
+            ...canisterInfo,
+          };
+        }
```
Comment (rocky-fleek) [inline source/Background/Controller.js:43]: for now this is good but we should use getMultipleCanisterInfo in the future.

**Review by rocky-fleek — APPROVED** [review]

**Pull request merged**